### PR TITLE
Rewrite the API docs

### PR DESCRIPTION
**Pull request by gcapizzi**
- Make the structure consistent
- Remove any information already present in the official docs
- Always link to the official docs
- Use the same order as the official docs
- Add missing endpoints
- Update outdated entries

This is hard to evaluate from the diff, but you can take a look at the [new version of the document](https://github.com/cloudfoundry/korifi/blob/api-docs-cleanup/docs/api.md).

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -1,468 +1,427 @@
...
+This document lists all the CF API endpoints supported by Korifi and their parameters.
 
-## Resources
+## Cloud Foundry V3 API
 
-### Root
+### [Apps](https://v3-apidocs.cloudfoundry.org/version/3.117.0/#apps)
 
-Docs: https://v3-apidocs.cloudfoundry.org/version/3.117.0/index.html#root
+#### [Create an app](https://v3-apidocs.cloudfoundry.org/version/3.117.0/#create-an-app)
```
Comment (tcdowney) [inline docs/api.md:9]: Nothing we can do about it, but a big gripe I have with the V3 API docs is that there is no way to link to the latest release.

File: docs/api.md [hunk 2]
```diff
@@ -1,468 +1,427 @@
...
-        "data": {
-          "guid": "7304bc3c-7010-11ea-8840-48bf6bec2d78"
-        }
-      }
-    }
-  }'
-```
+### [Manifests](https://v3-apidocs.cloudfoundry.org/version/3.117.0/#manifests)
+
+#### [Applying a manifest](https://v3-apidocs.cloudfoundry.org/version/3.117.0/#create-a-route)
```
Comment (akrishna90) [inline docs/api.md:134]: the link is incorrect. It need to point to - https://v3-apidocs.cloudfoundry.org/version/3.117.0/#apply-a-manifest-to-a-space

File: docs/api.md [hunk 3]
```diff
@@ -1,468 +1,427 @@
...
+-   `destinations[].protocol`
+
+#### [Remove destination for a route](https://v3-apidocs.cloudfoundry.org/version/3.117.0/#remove-destination-for-a-route)
 
 This endpoint is fully supported.
 
-### User Identity
+### [Service Instances](https://v3-apidocs.cloudfoundry.org/version/3.117.0/#service-instances)
+
+Korifi only supports user-provided service instances. [Fields](https://v3-apidocs.cloudfoundry.org/version/3.117.0/#fields) are not supported.
```
Comment (tcdowney) [inline docs/api.md:312]:
```suggestion
Korifi only supports user-provided service instances. Managed service operations and the [Fields](https://v3-apidocs.cloudfoundry.org/version/3.117.0/#fields) parameter are not supported.
```

File: docs/api.md (hunk 1 above)
Comment (gcapizzi) [reply docs/api.md:9]: Yeah 😞 I'll still update all links to point to the current latest!

File: docs/api.md (hunk 2 above)
Comment (gcapizzi) [reply docs/api.md:134]: Fixed, thanks!

File: docs/api.md (hunk 1 above)
Comment (tcdowney) [reply docs/api.md]: Wow that's great!

**Pull request merged**